File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    
    const dayToInt = function(day){
        let [y, m, d] = day.split(".").map(Number);
        y = (y-2000)*12*28;
        m = m*28;
        return y+m+d;
    }
    
    today = dayToInt(today);
    
    const term = {};
    terms.forEach(e => {
        var [k, v] = e.split(" ");
        term[k] = Number(v)*28;
    })
    
    privacies.forEach((priv, idx) => {
        var [date, type] = priv.split(" ");
        
        privDate = dayToInt(date) + term[type];
        
        if(today >= privDate) answer.push(idx+1);
    })
    return answer;
}

File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    public int dateToInt(String str){
        String[] ymd = str.split("[.]");
        int day = (Integer.parseInt(ymd[0])-2000)*12*28;
        day += Integer.parseInt(ymd[1])*28;
        day += Integer.parseInt(ymd[2]);
        return day;
    }
    
    
    public int[] solution(String today, String[] terms, String[] privacies) {
        List<Integer> answer = new ArrayList<>();
        int td = dateToInt(today);
        int pd = 0;
        String[] tmp = new String[2];
        Map<String, Integer> term = new HashMap<>();
        
        for(String s : terms){
            tmp = s.split(" ");
            term.put(tmp[0], Integer.parseInt(tmp[1]));
        }
        
        for(int i=0; i<privacies.length; i++){
            tmp = privacies[i].split(" ");
            pd = dateToInt(tmp[0]) + term.get(tmp[1])*28;
            if(pd <= td)  answer.add(i+1);
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}